source/DSCResources/DSC_CMSystemDiscovery/DSC_CMSystemDiscovery.psm1

```diff
@@ -0,0 +1,603 @@
+    Set-Location -Path "$($SiteCode):\"
+
+    $systemDiscovery = Get-CMDiscoveryMethod -Name ActiveDirectorySystemDiscovery -SiteCode $SiteCode
+
+    foreach ($prop in $systemDiscovery.Props)
+    {
+        switch ($prop.PropertyName)
+        {
+            'Settings'                          { $enabledStatus = ($prop.Value1 -eq 'Active') }
+            'Full Sync Schedule'                { $systemSchedule = $prop.Value1 }
+            'Enable Incremental Sync'           { $deltaEnabled = $prop.Value }
+            'Startup Schedule'                  { $systemDelta = $prop.Value1 }
+            'Enable Filtering Expired Logon'    { $lastLogonEnabled = $prop.Value }
+            'Days Since Last Logon'             { $lastLogon = $prop.Value }
+            'Enable Filtering Expired Password' { $lastPasswordEnabled = $prop.Value }
+            'Days Since Last Password Set'      { $lastPassword = $prop.Value }
+        }
+    }
+
+    $adContainersList = ($systemDiscovery.Proplists | Where-Object -FilterScript {$_.PropertyListName -eq 'AD Containers'}).Values
+    foreach ($line in $adContainersList)
+    {
+        if ($line -match 'LDAP://')
+        {
+            [array]$adContainerArray += $line
+        }
+    }
+
+    if ($deltaEnabled -eq 0)
+    {
+        $systemSchedule = $systemDelta
+        $systemDelta = $null
+    }
+
+    $scheduleConvert = ConvertTo-ScheduleInterval -ScheduleString $systemSchedule
+
+    if (-not [string]::IsNullOrEmpty($systemDelta))
+    {
+        $sDelta = Convert-CMSchedule -ScheduleString $systemDelta
+
+        if ($sDelta.HourSpan -eq 1)
+        {
+            $syncDelta = 60
+        }
+        else
+        {
+            $syncDelta = $sDelta.MinuteSpan
+        }
+    }
+
+    return @{
+        SiteCode                        = $SiteCode
+        Enabled                         = $enabledStatus
+        ScheduleInterval                = $scheduleConvert.Interval
+        ScheduleCount                   = $scheduleConvert.Count
+        EnableDeltaDiscovery            = $deltaEnabled
+        DeltaDiscoveryMins              = $syncDelta
+        EnableFilteringExpiredLogon     = $lastLogonEnabled
+        TimeSinceLastLogonDays          = $lastLogon
+        EnableFilteringExpiredPassword  = $lastPasswordEnabled
+        TimeSinceLastPasswordUpdateDays = $lastPassword
+        ADContainers                    = $adContainerArray
+    }
+}
+
+<#
+    .SYNOPSIS
+        This will set the desired state.
+
+    .PARAMETER SiteCode
+        Specifies the site code for Configuration Manager site.
+
+    .PARAMETER Enabled
+        Specifies the enablement of the System discovery method.
+
+    .PARAMETER EnableDeltaDiscovery
+        Indicates whether Configuration Manager discovers resources created or modified in AD DS
+        since the last discovery cycle. If you specify a value of $True for this parameter,
+        specify a value for the DeltaDiscoveryIntervalMinutes parameter.
```

Comment:
DeltaDiscoveryIntervalMinutes this parameter doesn't exist.

Reply:
updated to reflect the correct name